### PR TITLE
[Keyboard]Add X-Bows Nature  Keyboard

### PR DESCRIPTION
**Pull request by XBowsTech**
## Description

Add support for X-Bows Nature Ergonomic Mechanical Keyboard.
Nature, A 86 Keys Ergonomic Mechanical Keyboard with RGB backlight.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by XBowsTech** [comment]
I don't know why the PR check failed.Can someone explain it for me?
By the way, do I need to change anything?

**Review by zvecr — CHANGES_REQUESTED** [review]
One of the build is failing due to you providing an invalid json file.

**Comment by XBowsTech** [comment]
> One of the build is failing due to you providing an invalid json file.

Thank you very much! Under your reminder, I have found out the mistake and fixed it.

**Comment by XBowsTech** [comment]
> One of the build is failing due to you providing an invalid json file.

Are there any other problems with these documents? Can it be passed through?

File: keyboards/xbows/nature/config.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+#define DRIVER_ADDR_1 0b1110100
+#define DRIVER_ADDR_2 0b1110110
+#define DRIVER_ADDR_3 0b1110101
+#define DRIVER_COUNT 3
+#define DRIVER_1_LED_TOTAL 36
+#define DRIVER_2_LED_TOTAL 36
+#define DRIVER_3_LED_TOTAL 15
+#define DRIVER_LED_TOTAL (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL + DRIVER_3_LED_TOTAL)
+#endif
```
Comment (drashna) [inline keyboards/xbows/nature/config.h:63]:
```suggestion
#pragma once

#include "config_common.h"

#define VENDOR_ID 0xFEED
#define PRODUCT_ID 0x1226
#define DEVICE_VER 0x0001
#define MANUFACTURER X - BOWS
#define PRODUCT NATURE

#define MATRIX_ROWS 6
#define MATRIX_COLS 15
#define MATRIX_ROW_PINS { F7, F6, F5, F4, F1, F0 }
#define MATRIX_COL_PINS { B0, B1, B2, B3, B7, D2, D3, D5, D4, D6, D7, B4, B5, B6, C6 }
#define DIODE_DIRECTION COL2ROW

#ifdef RGB_MATRIX_ENABLE
#    define RGB_MATRIX_LED_PROCESS_LIMIT 18
#    define RGB_MATRIX_LED_FLUSH_LIMIT 16
#    define DEBOUNCE 3
#    define RGB_DISABLE_AFTER_TIMEOUT 0          // number of ticks to wait until disabling effects
#    define RGB_DISABLE_WHEN_USB_SUSPENDED true  // turn off effects when suspended
#    define RGB_MATRIX_KEYPRESSES
#    define RGB_MATRIX_MAXIMUM_BRIGHTNESS 200
#    define DISABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#    define DISABLE_RGB_MATRIX_BAND_SAT
#    define DISABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#    define DISABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#    define DISABLE_RGB_MATRIX_SPLASH
#    define DISABLE_RGB_MATRIX_MULTISPLASH
#    define DISABLE_RGB_MATRIX_SOLID_SPLASH
#    define DISABLE_RGB_MATRIX_SOLID_MULTISPLASH
#    define DISABLE_RGB_MATRIX_DIGITAL_RAIN

#    define DRIVER_ADDR_1 0b1110100
#    define DRIVER_ADDR_2 0b1110110
#    define DRIVER_ADDR_3 0b1110101
#    define DRIVER_COUNT 3
#    define DRIVER_1_LED_TOTAL 36
#    define DRIVER_2_LED_TOTAL 36
#    define DRIVER_3_LED_TOTAL 15
#    define DRIVER_LED_TOTAL (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL + DRIVER_3_LED_TOTAL)
#endif
```

File: keyboards/xbows/nature/nature.c [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+ }
+ __attribute__ ((weak))
+
+ void rgb_matrix_indicators_user(void)
+ {
+     if (IS_LED_ON(host_keyboard_leds(), USB_LED_CAPS_LOCK))
+     {
+         rgb_matrix_set_color(45, 0xFF, 0x00, 0x00);
+     }
+ }
```
Comment (drashna) [inline keyboards/xbows/nature/nature.c:167]:
```suggestion

void suspend_power_down_kb(void) {
    rgb_matrix_set_suspend_state(true);
    suspend_power_down_user();
}

void suspend_wakeup_init_kb(void) {
    rgb_matrix_set_suspend_state(false);
    suspend_wakeup_init_user();
}

 __attribute__ ((weak)) void rgb_matrix_indicators_user(void) {
    if (host_keyboard_led_state().caps_lock) {
        rgb_matrix_set_color(45, 0xFF, 0x00, 0x00);
    }
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/xbows/nature/config.h (hunk 1 above)
Comment (XBowsTech) [reply keyboards/xbows/nature/config.h:63]: “X-BOWS” should be used instead of “X - BOWS”

Comment (drashna) [reply keyboards/xbows/nature/config.h:63]: Apparently clang disagreed ... :D
That should be better I think

Comment (XBowsTech) [reply keyboards/xbows/nature/config.h:63]: As this is a trademark, it cannot be changed at will.

File: keyboards/xbows/nature/readme.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+
+Keyboard Maintainer: X-BOWS  
+Hardware Supported: X-BOWS Nature  
+Hardware Availability: [X-BOWS](https://x-bows.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make xbows/nature:default
+
```
Comment (Erovia) [inline keyboards/xbows/nature/readme.md:21]: Can you add "clear instructions on how to reset the board into bootloader mode" as per [the PR Checklist](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs)?

File: keyboards/xbows/nature/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+   * |---------------------------------------------------------------------------------------------------------------------------------|
+   * | Ctl |   A   |   S   |   D  |   F  |   G  |      Bksp      |    H  |    J   |   K   |   L   |   ;  |  '"  |    Enter    | PgDn |
+   * |---------------------------------------------------------------------------------------------------------------------------------|
+   * |Shift|   Z  |   X  |   C  |   V  |   B  |       Enter       |    N  |    M   |   ,  |   .   |  /?  | Shift|      |  Up  |
+   * |---------------------------------------------------------------------------------------------------------------------------------|
+   * |Ctrl | GUI |     Alter   |     Space   |   Ctrl   |   Shift  |     Space     |     Alter    |  FN  | Ctrl | Lft  |  Dn  |  Rig |
+   * |---------------------------------------------------------------------------------------------------------------------------------|
+   */
+  // Please note:The actual physical location of the PGUP key is at the end of the third line of the keyboard.(Just like the comment above)
```
Comment (Erovia) [inline keyboards/xbows/nature/keymaps/default/keymap.c:34]:
```suggestion
```

File: keyboards/xbows/nature/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  [0] = LAYOUT(
+		KC_ESC,  KC_F1,   KC_F2, KC_F3,  KC_F4,  KC_F5,   KC_F6,   KC_F7, KC_F8,  KC_F9,   KC_F10, KC_F11,  KC_F12,  KC_DEL,  KC_PSCR,\
+		KC_GRV,  KC_1,    KC_2,  KC_3,   KC_4,   KC_5,             KC_6,   KC_7,  KC_8,   KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,\
+		KC_TAB,  KC_Q,    KC_W,  KC_E,   KC_R,   KC_T,    KC_PGUP, KC_Y,   KC_U,  KC_I,   KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,\
```
Comment (Erovia) [inline keyboards/xbows/nature/keymaps/default/keymap.c:38]:
```suggestion
		KC_TAB,  KC_Q,    KC_W,  KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,  KC_I,   KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_PGUP,\
```

File: keyboards/xbows/nature/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+		KC_CAPS, KC_A,    KC_S,  KC_D,   KC_F,   KC_G,    KC_BSPC, KC_H,  KC_J,   KC_K,    KC_L,   KC_SCLN, KC_QUOT, KC_ENT,  KC_PGDN,\
+		KC_LSFT, KC_Z,    KC_X,  KC_C,   KC_V,   KC_B,    KC_ENT,  KC_N,  KC_M,   KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP,\
+		KC_LCTL, KC_LGUI, KC_LALT,       KC_SPC, KC_LCTL, KC_LSFT, KC_SPC,        KC_RALT, MO(1),  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),
+  [1] = LAYOUT(
+    RESET,     NK_TOGG, RGB_TOG, RGB_MOD,  RGB_HUI,  RGB_SAI,   RGB_SAD,  RGB_VAI,  RGB_VAD,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   EEP_RST,\
+    AG_TOGG,   KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,             KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_NLCK,\
+    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_HOME,  KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,\
```
Comment (Erovia) [inline keyboards/xbows/nature/keymaps/default/keymap.c:45]:
```suggestion
    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_HOME,\
```

File: keyboards/xbows/nature/nature.h [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include "quantum.h"
+#define LAYOUT( \
+	K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012, K013, K014,  \
+	K100, K101, K102, K103, K104, K105,       K107, K108, K109, K110, K111, K112, K113, K114,  \
+	K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212, K213, K214,  \
```
Comment (Erovia) [inline keyboards/xbows/nature/nature.h:21]:
```suggestion
	K200, K201, K202, K203, K204, K205, K207, K208, K209, K210, K211, K212, K213, K214, K206,  \
```

**Review by Erovia — CHANGES_REQUESTED** [review]
Please check and test the keymap and matrix modifications with an actual keyboard.
Such matrix wiring "quirks" should be masked from the users by the layout macro.

File: keyboards/xbows/nature/keymaps/default/keymap.c (hunk 5 above)
Comment (XBowsTech) [reply keyboards/xbows/nature/keymaps/default/keymap.c:37]: great！

File: keyboards/xbows/nature/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+		KC_CAPS, KC_A,    KC_S,  KC_D,   KC_F,   KC_G,    KC_BSPC, KC_H,  KC_J,   KC_K,    KC_L,   KC_SCLN, KC_QUOT, KC_ENT,  KC_PGDN,\
+		KC_LSFT, KC_Z,    KC_X,  KC_C,   KC_V,   KC_B,    KC_ENT,  KC_N,  KC_M,   KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP,\
+		KC_LCTL, KC_LGUI, KC_LALT,       KC_SPC, KC_LCTL, KC_LSFT, KC_SPC,        KC_RALT, MO(1),  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),
+  [1] = LAYOUT(
+    RESET,     NK_TOGG, RGB_TOG, RGB_MOD,  RGB_HUI,  RGB_SAI,   RGB_SAD,  RGB_VAI,  RGB_VAD,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   EEP_RST,\
+    AG_TOGG,   KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,             KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_NLCK,\
+    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_HOME,\
+    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_END,\
+    KC_TRNS,   UC_M_WI, UC_M_MA, UC_M_LN,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_VOLU,\
```
Comment (fauxpark) [inline keyboards/xbows/nature/keymaps/default/keymap.c:46]:
```suggestion
		KC_ESC,  KC_F1,   KC_F2, KC_F3,  KC_F4,  KC_F5,   KC_F6,   KC_F7, KC_F8,  KC_F9,   KC_F10, KC_F11,  KC_F12,  KC_DEL,  KC_PSCR,
		KC_GRV,  KC_1,    KC_2,  KC_3,   KC_4,   KC_5,             KC_6,   KC_7,  KC_8,   KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
		KC_TAB,  KC_Q,    KC_W,  KC_E,   KC_R,   KC_T,    KC_Y,    KC_U,  KC_I,   KC_O,    KC_P,   KC_LBRC, KC_RBRC, KC_BSLS, KC_PGUP,
		KC_CAPS, KC_A,    KC_S,  KC_D,   KC_F,   KC_G,    KC_BSPC, KC_H,  KC_J,   KC_K,    KC_L,   KC_SCLN, KC_QUOT, KC_ENT,  KC_PGDN,
		KC_LSFT, KC_Z,    KC_X,  KC_C,   KC_V,   KC_B,    KC_ENT,  KC_N,  KC_M,   KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP,
		KC_LCTL, KC_LGUI, KC_LALT,       KC_SPC, KC_LCTL, KC_LSFT, KC_SPC,        KC_RALT, MO(1),  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),
  [1] = LAYOUT(
    RESET,     NK_TOGG, RGB_TOG, RGB_MOD,  RGB_HUI,  RGB_SAI,   RGB_SAD,  RGB_VAI,  RGB_VAD,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   EEP_RST,
    AG_TOGG,   KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,             KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_NLCK,
    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_HOME,
    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,   KC_END,
    KC_TRNS,   UC_M_WI, UC_M_MA, UC_M_LN,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,   KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_VOLU,
```

File: keyboards/xbows/nature/readme.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+X-Bows was designed with three criteria in mind:
+*Enhancing the comfort and ergonomics of the keyboard
+*Limiting the learning curve for our new layout
+*Creating an attractive design
+
+
+Keyboard Maintainer: X-BOWS  
+Hardware Supported: X-BOWS Nature  
+Hardware Availability: [X-BOWS](https://x-bows.com/)
```
Comment (fauxpark) [inline keyboards/xbows/nature/readme.md:16]:
```suggestion
* Enhancing the comfort and ergonomics of the keyboard
* Limiting the learning curve for our new layout
* Creating an attractive design


* Keyboard Maintainer: X-BOWS
* Hardware Supported: X-BOWS Nature
* Hardware Availability: [X-BOWS](https://x-bows.com/)
```

File: keyboards/xbows/nature/readme.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
+# NATURE
+![NATURE](https://cdn.shopify.com/s/files/1/0014/7623/1237/files/xbows_1_720x_99574cc7-a547-4c1d-b847-482ec282f45b_540x.png)
+
+Nature, A 86 Keys Ergonomic Mechanical Keyboard with RGB backlight.
```
Comment (fauxpark) [inline keyboards/xbows/nature/readme.md:4]:
```suggestion
Nature, A 86 Keys Ergonomic Mechanical Keyboard with RGB backlight.

```

File: keyboards/xbows/nature/rules.mk [hunk 11]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (fauxpark) [inline keyboards/xbows/nature/rules.mk:6]:
```suggestion
```

File: keyboards/xbows/nature/rules.mk [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+BACKLIGHT_ENABLE = no
+MOUSEKEY_ENABLE = yes              # Mouse keys
+EXTRAKEY_ENABLE = yes              # Audio control and System control
+CONSOLE_ENABLE = no                # Console for debug
+COMMAND_ENABLE = no                # Commands for debug and configuration
+#SLEEP_LED_ENABLE = yes            # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes                  # USB Nkey Rollover
+AUDIO_ENABLE = no                  # Audio output
+RGB_MATRIX_ENABLE = yes            # Use RGB matrix
+NO_USB_STARTUP_CHECK = no          # Disable initialization only when usb is plugged in
```
Comment (fauxpark) [inline keyboards/xbows/nature/rules.mk:20]:
```suggestion
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = IS31FL3731
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/xbows/nature/rules.mk [hunk 13]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+# change yes to no to disable
+
```
Comment (fauxpark) [inline keyboards/xbows/nature/rules.mk:9]:
```suggestion
#
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/xbows/nature/rules.mk (hunk 13 above)
Comment (fauxpark) [inline keyboards/xbows/nature/rules.mk:9]:
```suggestion
#   change yes to no to disable
#
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/xbows/nature/readme.md (hunk 3 above)
Comment (Erovia) [reply keyboards/xbows/nature/readme.md:22]: I think this still needs to be added.

Comment (XBowsTech) [reply keyboards/xbows/nature/readme.md:22]: Flashing example for this keyboard:

    make xbows/nature:default:flash

Is that it?   I added.

**Review by fauxpark — APPROVED** [review]

**Review by Erovia — APPROVED** [review]

**Pull request merged**